From e743184e0cefef2bc73f870de594ca1bc815152c Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 4 Sep 2025 14:29:35 +0200
Subject: [PATCH 036/553] [docs] Fix \relates on QTEST_THROW_ON_{FAIL,SKIP}
 \macro

The other macros, e.g. QCOMPARE(), use \relates QTest. I used \relates
<QTest>, which puts these two macros on a page of their own, which was
not intended.

Fix by using the correct \relates target. This will break links to
https://doc.qt.io/qt-6/qtest-qttestlib-proxy.html, but we can't use
two \relates for the same entitiy (that's QTBUG-116182), so we have to
live with that.

Amends e769cf026e328ed7fff660c204ce6e55b80114e3.

Pick-to: 6.9 6.8
Change-Id: I4bb05b2f96e6e2195a5b281638b6a68cbe4c43af
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
Reviewed-by: Paul Wicking <paul.wicking@qt.io>
(cherry picked from commit 88dd5204075c7c0ff5402988f10b4bd11ad58fcb)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/testlib/qtestcase.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/testlib/qtestcase.cpp b/src/testlib/qtestcase.cpp
index 22c8fc5581d..7f07ac5213a 100644
--- a/src/testlib/qtestcase.cpp
+++ b/src/testlib/qtestcase.cpp
@@ -207,7 +207,7 @@ void Internal::maybeThrowOnSkip()
 /*!
     \since 6.8
     \macro QTEST_THROW_ON_FAIL
-    \relates <QTest>
+    \relates QTest
 
     When defined, QCOMPARE()/QVERIFY() etc always throw on failure.
     QTest::setThrowOnFail() then no longer has any effect.
@@ -216,7 +216,7 @@ void Internal::maybeThrowOnSkip()
 /*!
     \since 6.8
     \macro QTEST_THROW_ON_SKIP
-    \relates <QTest>
+    \relates QTest
 
     When defined, QSKIP() always throws. QTest::setThrowOnSkip() then no longer
     has any effect.
-- 
2.51.2

